rcgen/tests/generic.rs

```diff
@@ -518,3 +518,54 @@ mod test_csr {
 		assert_eq!(*params, csrp.params);
 	}
 }
+
+#[cfg(feature = "x509-parser")]
+mod test_subject_alternative_name_criticality {
+	use x509_parser::certificate::X509Certificate;
+	use x509_parser::extensions::X509Extension;
+	use x509_parser::{oid_registry, parse_x509_certificate};
+
+	use crate::util::default_params;
+
+	#[test]
+	fn with_subject_sans_not_critical() {
+		let (params, keypair) = default_params();
+		assert!(
+			!params
+				.distinguished_name
+				.iter()
+				.collect::<Vec<_>>()
+				.is_empty(),
+			"non-empty subject required for test"
+		);
```

Comment:
this is a little awkward, but it didn't seem worth adding API surface for, or moving the test into the same module to get at the `entries`.